Source/JavaScriptCore/inspector/remote/RemoteConnectionToTarget.h

```diff
@@ -106,6 +107,7 @@ class RemoteConnectionToTarget final : public ThreadSafeRefCounted<RemoteConnect
 
     ThreadSafeWeakPtr<RemoteControllableTarget> m_target WTF_GUARDED_BY_LOCK(m_targetMutex);
     bool m_connected { false };
+    std::atomic<bool> m_closed { false };
```

Comment:
actually i think we could just reuse `m_connected`, tho you'd have to make a few other adjustments in `RemoteConnectionToTarget::close` to set it back to `false`
```cpp
if (std::exchange(m_connected, false))
    target->disconnect(*this);
```

Reply:
It turns out we still might need to send messages between frontend and target before the connection is established and `m_connected` becomes `true`. I'm changing that state variable to `m_connectionState` to work around that instead.